pkg/daemon/update.go

```diff
@@ -1165,38 +1169,37 @@ func (dn *Daemon) deleteStaleData(oldIgnConfig, newIgnConfig *ign3types.Config)
 	return nil
 }
 
-// enableUnit enables a systemd unit via symlink
-func (dn *Daemon) enableUnit(unit ign3types.Unit) error {
-	// The link location
-	wantsPath := filepath.Join(wantsPathSystemd, unit.Name)
-	// sanity check that we don't return an error when the link already exists
-	if _, err := os.Stat(wantsPath); err == nil {
-		glog.Infof("%s already exists. Not making a new symlink", wantsPath)
-		return nil
-	}
-	// The originating file to link
-	servicePath := filepath.Join(pathSystemd, unit.Name)
-	err := renameio.Symlink(servicePath, wantsPath)
+// enableUnits enables a set of systemd units via systemctl, if any fail all fails.
+func (dn *Daemon) enableUnits(units []string) error {
+	args := append([]string{"enable"}, units...)
+	stdouterr, err := exec.Command("systemctl", args...).CombinedOutput()
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "enabling unit: %v", stdouterr)
 	}
-	glog.Infof("Enabled %s", unit.Name)
-	glog.V(2).Infof("Symlinked %s to %s", servicePath, wantsPath)
+	glog.Infof("Enabled systemd units: %v", units)
 	return nil
 }
 
-// disableUnit disables a systemd unit via symlink removal
-func (dn *Daemon) disableUnit(unit ign3types.Unit) error {
-	// The link location
-	wantsPath := filepath.Join(wantsPathSystemd, unit.Name)
-	// sanity check so we don't return an error when the unit was already disabled
-	if _, err := os.Stat(wantsPath); err != nil {
-		glog.Infof("%s was not present. No need to remove", wantsPath)
-		return nil
+// disableUnits disables a set of systemd units via systemctl, if any fail all fails.
+func (dn *Daemon) disableUnits(units []string) error {
+	args := append([]string{"disable"}, units...)
+	stdouterr, err := exec.Command("systemctl", args...).CombinedOutput()
+	if err != nil {
+		return errors.Wrapf(err, "disabling unit: %v", stdouterr)
 	}
-	glog.V(2).Infof("Disabling unit at %s", wantsPath)
+	glog.Infof("Disabled systemd units %v", units)
+	return nil
+}
 
-	return os.Remove(wantsPath)
+// presetUnit resets a systemd unit to its preset via systemctl
+func (dn *Daemon) presetUnit(unit ign3types.Unit) error {
+	args := []string{"preset", unit.Name}
+	_, err := exec.Command("systemctl", args...).CombinedOutput()
+	if err != nil {
+		return errors.Wrapf(err, "running preset on unit")
```

Comment:
What's the reasoning for not reporting stdout/stderr here?  I notice that you were doing that and then removed it.

Reply:
added back in, was testing something